### PR TITLE
refactor: remove context helper duplicate functions

### LINKED ISSUE #40: Refactor context helper functions
Remove duplicates and "shortcut" functions

### PR DESCRIPTION
**Pull request by xcoulon**
fixes #40

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: test/token/token.go [hunk 1]
```diff
@@ -123,33 +119,3 @@ func GenerateTokenWithClaims(claims map[string]interface{}) string {
...
-
-func ContextWithTokenAndRequestID() (context.Context, string, string, string) {
-	identityID := uuid.NewV4().String()
-	ctx, ctxToken := EmbedTokenInContext(identityID, uuid.NewV4().String())
-	ctx = tokencontext.ContextWithTokenManager(ctx, TokenManager)
-	reqID := uuid.NewV4().String()
-	ctx = client.SetContextRequestID(ctx, reqID)
-
-	return ctx, identityID, ctxToken, reqID
-}
```
Comment (nurali-techie) [inline test/token/token.go:151]: so this `ContextWithTokenAndRequestID()` is removed .. it's not useful ??

File: token/token_context.go [hunk 2]
```diff
@@ -1,6 +1,4 @@
-// Package token_context contains the code that extract token manager from the
-// context.
-package tokencontext
+package token
```
Comment (nurali-techie) [inline token/token_context.go:1]: make sure if there are any service using this currently then we need to adopt it with this change :+1:

**Review by nurali-techie — APPROVED** [review]
Overall looks good :+1:

File: test/token/token.go (hunk 1 above)
Comment (xcoulon) [reply test/token/token.go:151]: right, let me restore it: it's not used in `common`, but it has duplicates in other repos, that may use this one instead in the future.

File: token/token_context.go (hunk 2 above)
Comment (xcoulon) [reply token/token_context.go:1]: for now no one uses it directly, but it's also currently duplicated in other repos.

**Pull request merged**